src/parquet/column/properties.h

```diff
@@ -193,12 +266,38 @@ class PARQUET_EXPORT WriterProperties {
 
   ParquetVersion::type version() const { return parquet_version_; }
 
-  Encoding::type encoding(const std::shared_ptr<schema::ColumnPath>& path) const {
+  std::string created_by() const { return parquet_created_by_; }
+
+  std::vector<Encoding::type> encodings(
+      const std::shared_ptr<schema::ColumnPath>& path) const {
     auto it = encodings_.find(path->ToDotString());
     if (it != encodings_.end()) { return it->second; }
+    std::vector<Encoding::type> encodings(3);
+    encodings[2] = default_encoding_;
+    encodings[1] = default_dictionary_encoding_;
+    encodings[0] = default_level_encoding_;
+    return encodings;
```

Comment:
You can add a private method that initializes a default set of encodings and rearrange this method to consistently return a const reference